src/aws/osml/data_intake/image_processor.py

```diff
@@ -109,13 +107,40 @@ def generate_ovr_file(self, preview_size: int = 1024) -> str:
         :returns: Path to the generated overview file.
         """
         ovr_file = self.source_file + ".ovr"
+
+        existing_overviews = self.dataset.GetRasterBand(1).GetOverviewCount()
+
+        new_full_path = None
+
+        if existing_overviews > 0:
+            logger.info("Existing internal overviews found.")
+            directory_path = os.path.dirname(self.source_file)
+            full_filename = os.path.basename(self.source_file)
+
+            new_filename = f"{os.path.splitext(full_filename)[0]}_translated{os.path.splitext(full_filename)[1]}"
```

Comment:
Why are we doing this? If the image already has overviews we should just use them. We don't need to strip out the existing overviews and then recreate them later on.

Reply:
I brought this up, with the STAC Item design structure or format, we are required to upload an overview file to S3 bucket. That block of code ensure it can generate the overview file if internal exists and upload it to S3 bucket.